src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,46 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                var serviceProvider = ExtensionServiceProvider.Create(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
```

Comment:
Is there a reason you don't use the existing `ServiceHostInstance.ServiceProvider` and update it with the new assembly? The reason I am asking is that this should ensure that it can share common services to the extension if needed. Otherwise this won't happen

Reply:
Actually I would like to use ServiceHostInstance.ServiceProvider, what's the API to add a new assembly? Create is a static API.

Reply:
Fixed as you suggested